test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/DecorateChunksTestCodeGenerator.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Chunks;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+
+namespace Microsoft.AspNetCore.Razor.Test.CodeGenerators
+{
+    public class DecorateChunksTestCodeGenerator : CSharpCodeGenerator 
+    {
+        public enum GeneratorType
+        {
+            Add = 0,        // Add duplicates the first chunk in the list.
+            Clear = 1,      // Clear removes all chunks.
+            Default = 2     // Default does not modify chunks.
+        }
+
+        private readonly int _generatorType;
+
+        public DecorateChunksTestCodeGenerator(CodeGeneratorContext context, int type) : base(context)
+        {
+            _generatorType = type;
+        }
+
+        protected override void DecorateChunks(CodeGeneratorContext context, IList<Chunk> chunks)
+        {
+            switch (_generatorType)
+            {
+                case (int)GeneratorType.Add:
+                    var firstChunk = chunks.First();
+                    chunks.Add(firstChunk);
+                    break;
+                case (int)GeneratorType.Clear:
+                    chunks.Clear();
+                    break;
+                case (int)GeneratorType.Default:
```

Comment:
Why bother subclassing `CSharpCodeGenerator` if the overridden `DecorateChunks()` is a no-op? `CSharpCodeGenerator` isn't `abstract`.